tasks/us/census/tiger.py

```diff
@@ -385,6 +388,95 @@ def output(self):
+        return DownloadTiger(year=self.year)
+
+    def run(self):
+        session = current_session()
+        query = '''
+                CREATE TABLE {table_output} AS
+                SELECT *
+                FROM "{schema_input}".{table_input}
+                WHERE statefp10 = '{state}'
+                '''.format(schema_input='tiger{year}'.format(year=self.year),
+                           table_input=SUMLEVELS[self.geography]['table'],
+                           table_output=self.output().table,
+                           state=self.state)
+        session.execute(query)
+        session.commit()
+
+    def output(self):
+        return PostgresTarget('tiger{year}'.format(year=self.year),
+                              '{name}_state{state}'.format(name=SUMLEVELS[self.geography]['table'],
+                                                           state=self.state))
+
+
+class SimplifyGeoChunkByState(Task):
+    year = Parameter()
+    geography = Parameter()
+    state = Parameter()
+
+    def requires(self):
+        return SplitByState(year=self.year, geography=self.geography, state=self.state)
+
+    def run(self):
+        yield Simplify(schema=self.input().schema,
+                       table=self.input().tablename,
+                       table_id='.'.join(['tiger{year}'.format(year=self.year),
+                                          '{geo}_by_state'.format(geo=self.geography)]))
+
+    def output(self):
+        return PostgresTarget('tiger{year}'.format(year=self.year),
+                              '{name}_state{state}{suffix}'.format(name=SUMLEVELS[self.geography]['table'],
+                                                                   state=self.state,
+                                                                   suffix=SIMPLIFIED_SUFFIX))
+
+
+class SimplifyGeoByState(Task):
+    year = Parameter()
+    geography = Parameter()
+
+    def requires(self):
+        simplifications = {}
+        for state_code, _ in STATES.items():
+            simplifications[state_code] = SimplifyGeoChunkByState(year=self.year,
+                                                                  geography=self.geography,
+                                                                  state=state_code)
+        return simplifications
+
+    def run(self):
+        session = current_session()
+        for _, table in self.input().items():
+            query = '''
+                    CREATE TABLE IF NOT EXISTS {output} AS
+                    SELECT *
+                    FROM "{schema_input}".{table_input}
+                    WHERE 1=0
+                    '''.format(schema_input=table.schema,
+                               table_input=table.tablename,
+                               output=self.output().table)
+            session.execute(query)
+
+            query = '''
+                    INSERT INTO {output}
+                    SELECT *
+                    FROM "{schema_input}".{table_input}
```

Comment:
Why do you make it in two steps?

Reply:
These `create` + `insert` are inside a `for` loop that iterates over all the simplified tables. This inserts into the output table every iteration but creates the table only on the first one (`IF NOT EXISTS`)

Reply:
offline -> `input_table` has all the same columns. First create table creates a table with the output table not with the input table and then start with input